vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSlideController.java

```diff
@@ -0,0 +1,73 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import edu.asu.diging.vspace.core.model.IModule;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+import edu.asu.diging.vspace.core.services.impl.SlideManager;
+
+@Controller
+public class ExhibitionSlideController {
+
+	@Autowired
+	private IModuleManager moduleManager;
+
+	@Autowired
+	private SlideManager sildeManager;
+
+	@Autowired
+	private ISequenceManager sequenceManager;
+
+	@RequestMapping(value = "/exhibit/module/{moduleId}/sequence/{sequenceId}/slide/{slideId}", method = RequestMethod.GET)
+	public String slide(Model model, @PathVariable("slideId") String slideId,@PathVariable("moduleId") String moduleId, @PathVariable("sequenceId") String sequenceId) {
+		IModule module = moduleManager.getModule(moduleId);
+		model.addAttribute("module", module);
+		model.addAttribute("startSequenceId", module.getStartSequence().getId());
+		model.addAttribute("firstSlide", module.getStartSequence().getSlides().get(0).getId());
+		model.addAttribute("slides", sequenceManager.getSequence(sequenceId).getSlides());
+		model.addAttribute("currentSequenceId", sequenceId);
+		model.addAttribute("currentSlide", slideId);
+		model.addAttribute("currentSlideCon", sildeManager.getSlide(slideId));
+		model.addAttribute("startSequenceId", module.getStartSequence().getId());
+		return "module";
+	}
+
+	@RequestMapping(value = "/exhibit/module/{moduleId}/sequence/{sequenceId}/slide/{slideId}/slideShow", method = RequestMethod.GET)
+	public String slideShow(Model model, @PathVariable("slideId") String slideId,@PathVariable("moduleId") String moduleId, @PathVariable("sequenceId") String sequenceId) {
+		IModule module = moduleManager.getModule(moduleId);
+		model.addAttribute("module", module);
+		model.addAttribute("startSequenceId", module.getStartSequence().getId());
+		model.addAttribute("firstSlide", module.getStartSequence().getSlides().get(0).getId());
+		String nextSlideId="";
+		String prevSlideId="";
+		int index=0;
+		List<ISlide> slides=sequenceManager.getSequence(sequenceId).getSlides();
+		index=slides.indexOf(sildeManager.getSlide(slideId));
+		if(slides.size()>index+1) {
+			nextSlideId=slides.get(index+1).getId();
+		}
+		if(index>0) {
+			prevSlideId=slides.get(index-1).getId();
+		}
+		model.addAttribute("slides", slides);
+		model.addAttribute("currentSequenceId", sequenceId);
+		model.addAttribute("currentSlide", slideId);
+		model.addAttribute("nextSlide", nextSlideId);
+		model.addAttribute("prevSlide", prevSlideId);
+		model.addAttribute("currentSlideCon", sildeManager.getSlide(slideId));
+		model.addAttribute("startSequenceId", module.getStartSequence().getId());
+		model.addAttribute("numOfSlides", slides.size());
+		model.addAttribute("currentNumOfSlide", index+1);
+		return "slideShow";
+
+	}
```

Comment:
why are there two methods that bot seem to show a slide?

Reply:
One method is to load the home page of modules, just the first slide and the second method is to start slideShow as managing content on one jsp and handling visibilities is getting difficult.

Reply:
I'm confused, what is the slideshow in this case?

Reply:
there is a button called text version, on click of which the slideShow starts wherein a user can switch between all the slides in the sequence and on the same page user can come back to the module screen which has only the first slide of the start sequence.